dashboard/test/lib/api/v1/school_autocomplete_test.rb

```diff
@@ -20,4 +22,113 @@ class Api::V1::SchoolAutocompleteTest < ActiveSupport::TestCase
   test 'is zip search with invalid extended partial zip' do
     assert_equal false, Api::V1::SchoolAutocomplete.search_by_zip?('12345-ab')
   end
+
+  test 'search by unique name matches only 1 school' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('Chalkville', MAXIMUM_RESULTS, false)
+    assert_equal 1, search_results.count
+    assert search_results.detect {|school| school[:nces_id] == '10000200277'}
+  end
+
+  test 'search by zip returns match' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('98936', MAXIMUM_RESULTS, false)
+    # CHILDREN'S VILLAGE is in 98936
+    assert search_results.detect {|school| school[:nces_id] == '530537003179'}
+  end
+
+  test 'search by zip with no schools in it returns no match' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('10001', MAXIMUM_RESULTS, false)
+    assert 0, search_results.count
+  end
+
+  test 'search by school name also matches city' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('Albertville', MAXIMUM_RESULTS, false)
+    # ALBERTVILLE HIGH SCH
+    assert search_results.detect {|school| school[:nces_id] == '10000500871'}
+    # ALA AVENUE MIDDLE is in	ALBERTVILLE	AL
+    assert search_results.detect {|school| school[:nces_id] == '10000500870'}
+  end
+
+  test 'search by partial name matches school' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('Albert', MAXIMUM_RESULTS, false)
+    # ALBERT EINSTEIN ACADEMY ELEMENTARY
+    assert search_results.detect {|school| school[:nces_id] == '60000113717'}
+  end
+
+  test 'search by partial word matches school' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('Alb', MAXIMUM_RESULTS, false)
+    # ALBERT EINSTEIN ACADEMY ELEMENTARY
+    assert search_results.detect {|school| school[:nces_id] == '60000113717'}
+  end
+
+  test 'search by common abbreviation returns multiple matches' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('Sch', MAXIMUM_RESULTS, false)
+    assert_equal 8, search_results.count
+    # Alakanuk School
+    assert search_results.detect {|school| school[:nces_id] == '20000300216'}
+    # Sequoyah Sch   Chalkville Campus
+    assert search_results.detect {|school| school[:nces_id] == '10000200277'}
+  end
+
+  test 'punctuation in search is ignored' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('Pathways-College', MAXIMUM_RESULTS, false)
+    assert search_results.detect {|school| school[:nces_id] == '60001411746'}
+  end
+
+  test 'search by non-existent school name has no matches' do
+    search_results = Api::V1::SchoolAutocomplete.get_matches('MilhouseWuzHerr', MAXIMUM_RESULTS, false)
+    assert_equal 0, search_results.count
+  end
+
+  test 'new search by unique name matches only 1 school' do
```

Comment:
Thinking a bit more about this, your initial approach is better; it separates configuration from expectation. Configuration is to `use_new_search` or not, while expectation is input-to-output mapping. If later we have a new configuration, e.g. `use_new_search_v2`, we can add it without duplicating expectation logic.

Reply:
The "new search" intentionally has different functionality with different (hopefully more relevant) results for the same search, so I think making them separate tests is appropriate.